File: ruby/currying.rb
# Currying one argument at a time
# Partial application of arguments

add = proc { |x, y, z| x + y + z }

add5 = add.curry.call(5)

# equal to add.curry.call(5).call(6)
add5and6 = add5.call(6)

add5and6.call(10)  # => 21

File: haskell/currying.hs
-- Functions take single argument
-- Partial application of arguments

addOne :: Int -> Int -> Int
addOne x y = x + y

addTwo :: Int -> (Int -> Int)
addTwo x = \y -> x + y

addThree :: (Int -> (Int -> Int))
addThree = \x -> \y ->x + y





